### ISSUE TITLE
[Feature Request] Dynamically set Application name ("Playing ...")

### ISSUE THREAD

**Issue by VelocityRa** [discordapp/discord-rpc#375]
Rich Presence does not support dynamically setting a custom App Name (as in "Now Playing ..."). It instead takes the App Name from the Application Name from under https://discordapp.com/developers.

The vast majority of people don't click on profiles to see rich presence details, this would be very useful for music/movies etc app rich presence support.
It's the difference between 99% of people seeing ie. "Playing <song title/artist>" instead of "Playing Music" or such.

It's a feature that even MSN messenger supported.

---

This Issue is a continuation of #54 that got [inexplicably](https://github.com/discord/discord-rpc/issues/54#issuecomment-1276032383) closed.

**Comment by a-github-account** [comment]
![image](https://user-images.githubusercontent.com/77020751/208488411-33dbdf13-8ed0-42e1-821b-f8df527c0a0d.png)

At the very least, users should be able to change the name ("Last.fm" here) dynamically.

**Comment by rollingmoai** [comment]
The  [AliucordRN RichPresence plugin](https://github.com/amsyarasyiq/aliucordrn-plugins#richpresence) has the ability to do this:
![image](https://user-images.githubusercontent.com/75471378/210299240-fe10cd1c-0cb1-4a00-8561-37aa05fb98d7.png)
![image](https://user-images.githubusercontent.com/75471378/210299221-7e416928-c9e4-46e8-b595-de2c55d39953.png)

@amsyarasyiq any tips on how we can do this?

**Comment by Lachee** [comment]
> 

Can you actually edit the application name _while running_ tho? Is it doing it via the RPC or is it using a API call to the developer portal?
Also note custom clients, being outside of the TOS, dont always follow the rules when it comes to implementing features

edit: After quick testing I can confirm they must be doing something that normal RPC clients don't have access to (assuming they do allow you to change the name during runtime) as trying to send the activities they do result in RPC errors:
![error](https://i.lu.je/2023/VsDebugConsole_ftvUaCLTQh.png)

**Comment by RyloRiz** [comment]
> Can you actually edit the application name while running tho

It should be possible from what I know of the Discord RPC API. There is a field for `name` in the [Activity Structure](https://discord.com/developers/docs/topics/gateway-events#activity-object).

I would 100% support this being added. I have numerous use cases for this and it seems logical given that most people tend to only glance at quick status, which only displays the activity name. I do this myself and it is annoying that you need a different client ID for every different custom app, since Discord limits the amount of applications you can have.

**Comment by rollingmoai** [comment]
Another reference: [Vencord CustomRPC plugin](https://github.com/Vendicated/Vencord/blob/main/src/plugins/customRPC.tsx)

**Comment by amsryq** [comment]
No, it is not possible from a real application. The only reason why modified Discord clients could is because it manually set the application name, bypassing the default behavior (which is fetching the application name from the ID).

**Comment by J0nathan550** [comment]
> Another reference: [Vencord CustomRPC plugin](https://github.com/Vendicated/Vencord/blob/main/src/plugins/customRPC.tsx)

It is possible because you expose your discord token :(

**Comment by J0nathan550** [comment]
> Another reference: [Vencord CustomRPC plugin](https://github.com/Vendicated/Vencord/blob/main/src/plugins/customRPC.tsx)

It's like with Spotify, you can create account there and link discord, when you link discord, you expose your discord token, so that they can detect activity.

**Comment by Lachee** [comment]
Rich Presence doesn't expose your token to anyone.

**Comment by J0nathan550** [comment]
> Rich Presence doesn't expose your token to anyone.

Then how people is able to change the status to `Listening to` or `Watching` when it's not the part of RPC?

**Comment by WaGi-Coding** [comment]
any news on that?

**Comment by J0nathan550** [comment]
> any news on that?

In order to actually change statuses to those, you need to extract actual user token. And use something else than this library. For example discord bots allow that feature because we literally parse the token of bot (like a user). And then using API calling specific code to change status to whatever you like. So in here, in this library, there will be no ability to add this feature unless to add the authentication, extracting user token and etc. To allow such functionality. This library is perfect for making "Game" activity.
